Makefile.dep

```diff
@@ -404,7 +404,7 @@ ifneq (,$(filter newlib,$(USEMODULE)))
     USEMODULE += newlib_syscalls_default
   endif
   ifeq (,$(filter stdio_rtt stdio_cdc_acm,$(USEMODULE)))
-    USEMODULE += stdio_uart
+    FEATURES_PROVIDED += stdio_uart
```

Comment:
I think this is forbidden.

Reply:
Where and why?

Reply:
Features should only be provided in Makefile.features files. Makefile.dep files are here to do the resolution of dependencies given the provided features. If we start mixing the concepts we will end up in more mess.

This PR doesn't work `make info-boards-supported`, so Murdock will fail in any case:
```
make -C tests/gnrc_ipv6_ext_frag info-boards-supported --no-print-directory | tr " " "\n" | grep thingy52
thingy52
```
Normally the result should be empty if the board is excluded.